packages/react/src/components/ComboBox/ComboBox.tsx

```diff
@@ -817,16 +829,21 @@ const ComboBox = forwardRef(
           type,
           selectedItem: newSelectedItem,
         });
+        if (isManualClearingRef.current) {
+          return;
+        }
         if (
           type === useCombobox.stateChangeTypes.ItemClick &&
-          !isEqual(selectedItemProp, newSelectedItem)
+          !isEqual(selectedItemProp, newSelectedItem) &&
+          newSelectedItem !== undefined
         ) {
           onChange({ selectedItem: newSelectedItem });
         }
         if (
           (type === useCombobox.stateChangeTypes.FunctionSelectItem ||
             type === useCombobox.stateChangeTypes.InputKeyDownEnter) &&
-          !isEqual(selectedItemProp, newSelectedItem) // Only fire if there's an actual change
+          !isEqual(selectedItemProp, newSelectedItem) &&
+          newSelectedItem !== undefined
```

Comment:
https://stackoverflow.com/a/2703122/11106650

```suggestion
          typeof newSelectedItem !== 'undefined'
```